prov/util/src/util_wait.c

```diff
@@ -273,6 +273,9 @@ static int util_wait_fd_run(struct fid_wait *wait_fid, int timeout)
 			return -FI_ETIMEDOUT;
 
 		ret = fi_epoll_wait(wait->epoll_fd, ep_context, 1, timeout);
+		if (ret > 0)
+			return FI_SUCCESS;
```

Comment:
The issue here is that eq_sread is returning even when there are no pending events yet in eq. 

For tcp case, wait's epoll only checks for events on socket level. Based on those we can't return eq_sread call, But that is happening here.